### ISSUE TITLE
Option to display notes at the bottom or at the top

### ISSUE THREAD

**Issue by vince-from-nice** [nextcloud/notes#1027]
Before, when I opened an existing note, it was displayed at the top of the content. Therefore most of my notes are designed like that (ie. recent or important lines are on the top). 

But from the latest update the notes are now displayed from the bottom. That forces me to scroll up every time I open a note. 

An option to select from where note contents are displayed would be very useful. 

Thanks, Vincent.

**Comment by juliushaertl** [comment]
I think this is a good point to bring up for discussion where to focus on a document. With the text editor we seem to indeed always focus the cursor at the end. Ideally we probably remember the users last position, but this might of course be tricky if the file has been changed outside in the meantime. 

cc @jancborchardt @nimishavijay for design input
@mejo- @max-nextcloud for Text/Collectives.

**Comment by nimishavijay** [comment]
Great point and I agree, remembering the position of the cursor seems like it is the best way to go :) Would it be possible to calculate the changes in the document so that even if if it has been edited by another person, the position of the cursor remains at the same character? That would be the most ideal solution. 

As a quick fix, we could maybe already implement remembering the cursor position without factoring in collaboration in the notes app only, since it is mostly for personal notes?

**Comment by vince-from-nice** [comment]
Ok if I use the old "edit mode" as default view,  the cursor is still set at the top of the notes.  It's good workaround for me. Btw there is no way to switch to the new rich text editor (without setting it as the default view mode) ?

But yes, if the cursor position could be saved for each note it should very useful. I update the title of the issue.

**Comment by max-nextcloud** [comment]
I remember seeing different behavior here in collectives. For some time it would scroll to the last image on a page because images were always centered after being added - even when they preexisted in the document. I think that has been fixed.

I agree that last cursor position would be best - if it's not available i think we should default to the start of the page.

**Comment by mejo-** [comment]
> I agree that last cursor position would be best - if it's not available i think we should default to the start of the page.

Yep, fully agree on that. It means though that we have to persist the cursor position in DB for each user + document combination. Or would it be better to persist it in browser storage?

**Comment by max-nextcloud** [comment]
> Yep, fully agree on that. It means though that we have to persist the cursor position in DB for each user + document combination. Or would it be better to persist it in browser storage?

Both seem good to me. Implementation wise I think browser storage might be easier as it would not require a server side api.

UX wise i am not sure. I might open the same file on my phone and my laptop. Not sure if i would want the two cursor positions to influence each other. But reading something at work on the laptop and then continuing while commuting seems like a reasonable usage pattern. It would also allow picking this up from inside the android app. So I think I'd go with the database.

**Comment by juliushaertl** [comment]
Maybe browser storage could be a good start keep the effort reasonable. That way we could proof also scenarios where the document had changed in the meantime. Persisting across devices could be a second step then.

**Comment by juliushaertl** [comment]
This has been implemented with https://github.com/nextcloud/text/pull/4540
